teammapper-frontend/mmp/src/map/handlers/history.ts

```diff
@@ -54,17 +54,40 @@ export default class History {
 
             this.save()
 
-            if(notifyWithEvent) this.map.events.call(Event.create, this.map.dom)
+            if (notifyWithEvent) this.map.events.call(Event.create, this.map.dom)
         } else if (this.checkSnapshotStructure(snapshot)) {
             const previousData = this.map.export.asJSON()
 
+            // Find all nodes where we've set hasHiddenChildNodes in the previous map
```

Comment:
maybe apply refactoring: extract method, as this is a lot of logic only dedicating to the hiding of nodes.